### ISSUE TITLE
`selectOptions` with Angular

### ISSUE THREAD

**Issue by timdeschryver** [testing-library/user-event#154]
👋 first of all, great idea to create this library!

I recently found out about this library and was looking to use it with the Angular testing library. 
To be able to do that I had to wrap the user events with some angular logic (has to do with the Angular change detection), so now I'm exposing these wrapped user events via the `render` function. But there's a problem with `selectOptions`. To work in Angular,  it also needs to fire a change event (instead of just clicking on the option element). 

Now my question, should I provide these user events in the Angular testing library myself, trying to keep a similar API? Or should we make the change in this library to be able to use it with Angular?

I'm fine either way, I just want to hear your opinions before I go through with the implementation 🙂

**Comment by timdeschryver** [comment]
Fyi, I went ahead and implemented the `user-events` in the angular testing library. This, to be able to call Angular's change detection with each user action. 



https://github.com/testing-library/angular-testing-library/pull/37

**Comment by Gpx** [comment]
Hey, I think I missed the notification for this issue. It probably makes sense to implement it there. In the long run, this project should be included with the other main libraries too.

**Comment by timdeschryver** [comment]
👍 Thanks for the feedback @Gpx !
I'll implement them in the Angular library, and stick to the current API (or as close as possible)

**Comment by Gpx** [comment]
Hey, we just merged #158 which I think might solve this issue. Do you mind taking a look at it?

**Comment by timdeschryver** [comment]
I haven't verified it, but this seems like it would solve the issue.
I'll continue to implement the function myself in the angular library (in the next coming week) with the implementation from the referenced PR. 

I think I will continue to implement the events myself, to be able to fire Angular's change detection on every fired event.

**Comment by timdeschryver** [comment]
@Gpx I'm currently tried it in Angular, and for us it's not enough to just fire the event, we also have to provide the target's value:

```
fireEvent.change(select, { target: { value } });
```

**Comment by timdeschryver** [comment]
You can completely ignore the above comment, it does work.

**Comment by timdeschryver** [comment]
I did the implementation for Angular, but I deviated from the the implementation without knowing it. 

Instead of selection the options based on the value (`valArray.includes(opt.value)`), I used the `getByText` queries to select the option. Is there a a specific reason to use the option's value, and should I change my implementation?

While implementing it, I think I also found a bug with a multiple select. The previous value isn't being reset, for example:

```js
userEvent.selectOptions(getByTestId("element"), ["2"]);
userEvent.selectOptions(getByTestId("element"), ["1", "3"]);
```

Results in value 1, 2 and 3 being selected.
